Platform api changes to support phase 2 of Dockerfiles implementation (#320)
buildpacks/spec: platform.md

```

##### Inputs
| Input           | Environment Variable | Default Value            | Description                                                                 |
|-----------------|----------------------|--------------------------|-----------------------------------------------------------------------------|
| `<analyzed>`    | `CNB_ANALYZED_PATH`  | `<layers>/analyzed.toml` | Path to output analysis metadata (see [`analyzed.toml`](#analyzedtoml-toml) |

> | `<build-image>` | `CNB_BUILD_IMAGE`    |                          | Reference to the current build image in an OCI registry                     |

Author: ```suggestion
| `<build-image>` | `CNB_BUILD_IMAGE`    |                          | Reference to the current build image in an OCI registry (if used `<kaniko-dir>` must be provided)                     |
```

| `<cache-dir>`   | `CNB_CACHE_DIR`      |                          | Path to a cache directory                                                   |
| `<cache-image>` | `CNB_CACHE_IMAGE`    |                          | Reference to a cache image in an OCI registry                               |
| `<gid>`         | `CNB_GROUP_ID`       |                          | Primary GID of the build image `User`                                       |
| `<group>`       | `CNB_GROUP_PATH`     | `<layers>/group.toml`    | Path to group definition (see [`group.toml`](#grouptoml-toml))              |
| `<layers>`      | `CNB_LAYERS_DIR`     | `/layers`                | Path to layers directory                                                    |
| `<log-level>`   | `CNB_LOG_LEVEL`      | `info`                   | Log Level                                                                   |
| `<uid>`         | `CNB_USER_ID`        |                          | UID of the build image `User`                                               |

> | `<skip-layers>` | `CNB_SKIP_LAYERS`    | `false`                  | Do not perform [layer restoration](#layer-restoration)                      |

Author: ```suggestion
| `<skip-layers>` | `CNB_SKIP_LAYERS`    | `false`                  | Do not perform [layer restoration](#layer-restoration)                      |
|`<kaniko-dir>`| | | Kaniko directory (must be `/kaniko`) |
```

| `/dev/stdout`                               | Logs (info)                                                                                                                               |
| `/dev/stderr`                               | Logs (warnings, errors)                                                                                                                   |
| `<layers>/<buidpack-id>/store.toml`         | Persistent metadata (see data format in [Buildpack Interface Specification](buildpack.md))                                                |
| `<layers>/<buidpack-id>/<layer>.toml`       | Files containing the layer content metadata of each analyzed layer (see data format in [Buildpack Interface Specification](buildpack.md)) |
| `<layers>/<buidpack-id>/<layer>.sbom.<ext>` | Files containing the Software Bill of Materials for each analyzed layer (see [Buildpack Interface Specification](buildpack.md))           |
| `<layers>/<buidpack-id>/<layer>/*`.         | Restored layer contents                                                                                                                   |

> | `/kaniko/cache`                             | Kaniko cache contents                                                                                                                     |

Author: ```suggestion
| `<kaniko-dir>/cache`                             | Kaniko cache contents                                                                                                                     |
```

| `<build-image>` | `CNB_BUILD_IMAGE`    |                          | Reference to the current build image in an OCI registry                     |
| `<cache-dir>`   | `CNB_CACHE_DIR`      |                          | Path to a cache directory                                                   |
| `<cache-image>` | `CNB_CACHE_IMAGE`    |                          | Reference to a cache image in an OCI registry                               |
| `<gid>`         | `CNB_GROUP_ID`       |                          | Primary GID of the build image `User`                                       |
| `<group>`       | `CNB_GROUP_PATH`     | `<layers>/group.toml`    | Path to group definition (see [`group.toml`](#grouptoml-toml))              |
| `<layers>`      | `CNB_LAYERS_DIR`     | `/layers`                | Path to layers directory                                                    |
| `<log-level>`   | `CNB_LOG_LEVEL`      | `info`                   | Log Level                                                                   |
| `<uid>`         | `CNB_USER_ID`        |                          | UID of the build image `User`                                               |

> | `<skip-layers>` | `CNB_SKIP_LAYERS`    | `false`                  | Do not perform [layer restoration](#layer-restoration)                      |

Author: ```suggestion
| `<skip-layers>` | `CNB_SKIP_LAYERS`    | `false`                  | Do not restore any data to buildpack layer directories                      |
```

- For each buildpack in `<group>`, if persistent metadata for that buildpack exists in the analysis metadata, lifecycle MUST write a toml representation of the persistent metadata to `<layers>/<buildpack-id>/store.toml`
- **If** `<skip-layers>` is `true` the lifecycle MUST NOT perform layer restoration.
- **Else** the lifecycle MUST perform [layer restoration](#layer-restoration) for any app image layers or cached layers created by any buildpack present in the provided `<group>`.
- When the provided `<group>` contains image extensions (**[experimental](#experimental-features)**), the lifecycle:

>   - MUST record the digest reference to the provided `<build-image>` in `<analyzed>`

Reviewer: Should we put the `build-image` key in the example at the bottom?

Author: Good catch @jabrown85 - added in 4864b74

>   - MUST copy the OCI manifest and config file for `<build-image>` to `/kaniko/cache`

Author: ```suggestion
  - MUST copy the OCI manifest and config file for `<build-image>` to `<kaniko>/cache`
```

| `<plan>`             | `CNB_PLAN_PATH`        | `<layers>/plan.toml`     | Path to resolved build plan (see [`plan.toml`](#plantoml-toml))                                 |
| `<platform>`         | `CNB_PLATFORM_DIR`     | `/platform`              | Path to platform directory                                                                      |
| `<uid>`              | `CNB_USER_ID`          |                          | UID of the build image `User`                                                                   |

##### Outputs

In addition to the outputs enumerated below, outputs produced by `extender` include those produced by `builder` - as the lifecycle will run the `build` phase after extending the build image. When using the `extender` platforms MUST skip the `builder` and proceed to the `exporter`.

| Output          | Description                            |
|-----------------|----------------------------------------|
| [exit status]   | (see Exit Code table below for values) |
| `/dev/stdout`   | Logs (info)                            |
| `/dev/stderr`   | Logs (warnings, errors)                |

> | `/kaniko/cache` | Kaniko cache contents                  |

Author: ```suggestion
| `<kaniko-dir>/cache` | Kaniko cache contents                  |
```

| `<kaniko-cache-ttl>` | `CNB_KANIKO_CACHE_TTL` | 2 weeks               | Kaniko cache TTL                                                                                |
| `<layers>`           | `CNB_LAYERS_DIR`       | `/layers`             | Path to layers directory                                                                        |
| `<log-level>`        | `CNB_LOG_LEVEL`        | `info`                | Log Level                                                                                       |
| `<plan>`             | `CNB_PLAN_PATH`        | `<layers>/plan.toml`  | Path to resolved build plan (see [`plan.toml`](#plantoml-toml))                                 |
| `<platform>`         | `CNB_PLATFORM_DIR`     | `/platform`           | Path to platform directory                                                                      |
| `<uid>`              | `CNB_USER_ID`          |                       | UID of the build image `User`                                                                   |

##### Outputs

> In addition to the outputs enumerated below, outputs produced by `extender` include those produced by `builder` - as the lifecycle will run the `build` phase after extending the build image.

Author: ```suggestion
In addition to the outputs enumerated below, outputs produced by `extender` include those produced by `builder` - as the lifecycle will run the `build` phase after extending the build image. When using the `extender` platforms MUST skip the `builder` and proceed to the `exporter`. 
```

|----------------------|------------------------|-----------------------|-------------------------------------------------------------------------------------------------|
| `<app>`              | `CNB_APP_DIR`          | `/workspace`          | Path to application directory                                                                   |
| `<buildpacks>`       | `CNB_BUILDPACKS_DIR`   | `/cnb/buildpacks`     | Path to buildpacks directory (see [Buildpacks Directory Layout](#buildpacks-directory-layout))  |
| `<generated>`        | `CNB_GENERATED_DIR`    | `<layers>/generated`  | (**[experimental](#experimental-features)**) Path to directory containing generated Dockerfiles |
| `<gid>`              | `CNB_GROUP_ID`         |                       | Primary GID of the build image `User`                                                           |
| `<group>`            | `CNB_GROUP_PATH`       | `<layers>/group.toml` | Path to group definition (see [`group.toml`](#grouptoml-toml))                                  |

> | `<image>`            |                        |                       | Digest reference to an image to extend                                                          |

Author: ```suggestion
| `<image>`            |                        |                       | OCI layout path of an image to extend                                                          |
```
